tests/integration/test_charm.py

```diff
@@ -35,20 +35,36 @@ async def test_build_and_deploy(ops_test, charm_under_test):
 
 async def test_relate(ops_test):
     await asyncio.gather(
-        ops_test.model.add_relation(PROM_NAME, APP_NAME),
-        ops_test.model.add_relation(APP_NAME, SPRING_NAME),
+        ops_test.model.add_relation(
+            f"{PROM_NAME}:metrics-endpoint", f"{APP_NAME}:metrics-endpoint"
+        ),
+        ops_test.model.add_relation(
+            f"{APP_NAME}:configurable-scrape-jobs", f"{ZINC_NAME}:metrics-endpoint"
+        ),
     )
     await ops_test.model.wait_for_idle(status="active")
 
 
 async def test_alert_rules_exist(ops_test):
     rules = await get_prometheus_rules(ops_test=ops_test, app_name=PROM_NAME, unit_num=0)
-    assert len(rules) > 0, "No alert rules are present even though spring music is related"
+    assert len(rules) > 0, "No alert rules are present even though zinc is related"
 
 
 async def test_multiple_workloads_alert_rules(ops_test):
     old_rules = await get_prometheus_rules(ops_test=ops_test, app_name=PROM_NAME, unit_num=0)
-    await ops_test.model.add_relation(APP_NAME, "spring-music2")
+    await ops_test.model.add_relation(APP_NAME, "zinc2")
     await ops_test.model.wait_for_idle(status="active")
     new_rules = await get_prometheus_rules(ops_test=ops_test, app_name=PROM_NAME, unit_num=0)
     assert len(new_rules) > len(old_rules), "Additional workload instance did not add alert rules"
+
+
+async def test_non_leader_units_set_waiting_status(ops_test):
+    await ops_test.model.applications[APP_NAME].scale(scale=2)
+    await ops_test.model.block_until(lambda: len(ops_test.model.applications[APP_NAME].units) == 2)
+    await ops_test.model.wait_for_idle()
```

Comment:
Should probably use `wait_for_idle` with `exact units`.

Reply:
I agree this would be a better way than the `block_until` above.

Reply:
`exact_units` waits for idle on that many units for each of the applications. This is not the present use case since all applications do not have the same number of units.